crypto/keys/hd/hdpath.go

```diff
@@ -55,13 +54,99 @@ func NewParams(purpose, coinType, account uint32, change bool, addressIdx uint32
 	}
 }
 
+// Parse the BIP44 path and unmarshal into the struct.
+// nolint: gocyclo
+func NewParamsFromPath(path string) (*BIP44Params, error) {
+	spl := strings.Split(path, "/")
+	if len(spl) != 5 {
+		return nil, fmt.Errorf("path length is wrong. Expected 5, got %d", len(spl))
+	}
+
+	if spl[0] != "44'" {
+		return nil, fmt.Errorf("first field in path must be 44', got %v", spl[0])
+	}
+
+	if !isHardened(spl[1]) || !isHardened(spl[2]) {
+		return nil,
+			fmt.Errorf("second and third field in path must be hardened (ie. contain the suffix ', got %v and %v", spl[1], spl[2])
+	}
+	if isHardened(spl[3]) || isHardened(spl[4]) {
+		return nil,
+			fmt.Errorf("fourth and fifth field in path must not be hardened (ie. not contain the suffix ', got %v and %v", spl[3], spl[4])
+	}
+
+	purpose, err := hardenedInt(spl[0])
+	if err != nil {
+		return nil, err
+	}
+	coinType, err := hardenedInt(spl[1])
+	if err != nil {
+		return nil, err
+	}
+	account, err := hardenedInt(spl[2])
+	if err != nil {
+		return nil, err
+	}
+	change, err := hardenedInt(spl[3])
+	if err != nil {
+		return nil, err
+	}
+	if !(change == 0 || change == 1) {
+		return nil, fmt.Errorf("change field can only be 0 or 1")
+	}
+
+	addressIdx, err := hardenedInt(spl[4])
+	if err != nil {
+		return nil, err
+	}
+
+	return &BIP44Params{
+		purpose:    purpose,
+		coinType:   coinType,
+		account:    account,
+		change:     change > 0,
+		addressIdx: addressIdx,
+	}, nil
+}
+
+func hardenedInt(field string) (uint32, error) {
+	field = strings.TrimSuffix(field, "'")
+	i, err := strconv.Atoi(field)
+	if err != nil {
+		return 0, err
+	}
+	if i < 0 {
+		return 0, fmt.Errorf("fields must not be negative. got %d", i)
+	}
+	return uint32(i), nil
```

Comment:
what if `i` is int64 on x64? possible overflow?

maybe use `ParseInt(field, 10, 32)` instead of `Atoi`?